datalad/support/gitrepo.py

```diff
@@ -3409,8 +3428,12 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
             # the config has changed too
             self.config.reload()
             # need to include .gitmodules in what needs saving
-            status[self.pathobj.joinpath('.gitmodules')] = dict(
-                type='file', state='modified')
+            f = self.pathobj.joinpath('.gitmodules')
+            # add/modify 'status' for it in-place and thus need to adjust
+            # status_state as well
+            status_state['modified_or_untracked'][f] = \
+                status_state['modified'][f] = \
```

Comment:
May be I am missing something, but:

1. Comment is confusing since you stopped modifying `status`, apparently.
2. I think you should keep `status` and `status_state` in sync. Otherwise this will become troublesome for future changes down the road.

```suggestion
                status_state['modified'][f] = \
                status[f] = \
```

Reply:
> 1. Comment is confusing since you stopped modifying `status`, apparently.

thanks for catching this -- it is leftover from original implementation of the PR where it was needed, removing now, and thus re 2.: see above `del status  # to ensure it is no longer used` at https://github.com/datalad/datalad/pull/6581/files#diff-bfc1c47d032fc72b15b6a79db3d49e4c4334cf96318c5efe48a3403744b32fb1R3336

Reply:
Right. Forgot the `del` further up.